hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataResolver.java

```diff
@@ -81,4 +85,75 @@ static void maybeAddDefaultField(
             tableFields.add(new MapTableField(fieldName, type, true, QueryPath.create(fieldName)));
         }
     }
+
+    /**
+     * If {@code __key}/{@code this} is the only key/value field and has a custom type with
+     * nonnull metadata, resolve the schema ID from the type. Otherwise, return {@code orElse()}.
```

Comment:
could you positively describe what cases are covered by `orElse`? "is the only key/value field"? "there are no fields"? "this is UDT"?